wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt

```diff
@@ -65,5 +70,74 @@ class AnyMessageTest {
     assertEquals(hex, usesAny.encodeByteString().hex())
     assertEquals(usesAny, UsesAny.ADAPTER.decode(hex.decodeHex()))
   }
+
+  @Test fun decodingWithMissingTypeUrl() {
+    val hex = "0a0412020803120412020803120412020804"
+
+    val usesAny = UsesAny.ADAPTER.decode(hex.decodeHex())
+    assertEquals("", usesAny.just_one!!.typeUrl)
+    assertEquals(2, usesAny.many_anys.size)
+    for (manyAny in usesAny.many_anys) {
+      assertEquals("", manyAny.typeUrl)
+    }
+  }
+
+  @Test fun unpackWithWrongTypeUrlThrow() {
+    val usesAny = UsesAny(
+        just_one = AnyMessage.pack(OneField(opt_int32 = 3)),
+        many_anys = listOf()
+    )
+
+    try {
+      usesAny.just_one!!.unpack(NoFields.ADAPTER)
+      fail()
+    } catch (expected: IllegalStateException) {
+      assertEquals("type mismatch: type.googleapis.com/squareup.protos.kotlin.edgecases.OneField " +
+          "!= type.googleapis.com/squareup.protos.kotlin.edgecases.NoFields", expected.message)
+    }
+  }
+
+  @Test fun unpackOrNullWithWrongTypeUrlReturnsNull() {
+    val usesAny = UsesAny(
+        just_one = AnyMessage.pack(OneField(opt_int32 = 3)),
+        many_anys = listOf()
+    )
+
+    assertNull(usesAny.just_one!!.unpackOrNull(NoTypeUrlMessage.ADAPTER))
+  }
+
+  @Test fun packWithoutATypeUrlThrows() {
+    try {
+      AnyMessage.pack(NoTypeUrlMessage())
+      fail()
+    } catch (expected: IllegalStateException) {
+      assertEquals("recompile class com.squareup.wire.AnyMessageTest.NoTypeUrlMessage to use " +
+          "it with AnyMessage",
+          // We cover all platform cases.
```

Comment:
I don’t follow this replace . . .  why not change the expectation?

Reply:
In JVM, we get `com.squareup.wire.AnyMessageTest$NoTypeUrlMessage`, we native we get `com.squareup.wire.AnyMessageTest.NoTypeUrlMessage`

Reply:
and JS has `NoTypeUrlMessage` only without the package name...

Reply:
Went with `assertTrue(expected.message!!.endsWith("NoTypeUrlMessage to use it with AnyMessage"))`

Reply:
multiplatform was the part I hadn’t anticipated!